service/src/main/resources/api/service_openapi.yaml

```diff
@@ -2386,6 +2456,75 @@ components:
       items:
         $ref: "#/components/schemas/ExportModel"
 
+    ResourceType:
+      type: string
+      enum: [ "STUDY", "COHORT", "REVIEW", "CONCEPT_SET" ]
+
+    Resource:
+      type: object
+      properties:
+        type:
+          $ref: "#/components/schemas/ResourceType"
+        studyId:
+          $ref: "#/components/schemas/StudyIdV2"
```

Comment:
Perhaps these should reference objects for each type rather than ids. For now most would just contain their ids but then you wouldn't need the `additionalInfo` grab bag.

Reply:
I'm not sure that would solve the `additionalInfo` bag of optional properties, since that depends on `ActionType`, not `ResourceType` (e.g. the backend would set `additionalInfo.exportType` for the `EXPORT_COHORT` `ActionType`, but not for `CREATE_COHORT`).

Definitely open to suggestions on how to better return information that is not universal for all action types. I considered making them not `required` properties of the top-level `ActivityLogEntry` object, but thought that made it harder to see what was common for all log entries. Open to opinions on that though.

Reply:
Ah I see. I think one object per `ActionType` is a lot of repetition of properties (e.g. all will have `studyId`). I'd prefer to just define them once.

I could also make `additionalInfo` a generic property map that action types can fill in as necessary. Upside is we don't need to change the API significantly to add a new action type (just a new `ActionType` enum value). Thoughts?

Reply:
Per offline discussion, we think the current structure of `additionalInfo` will best support the filtering use case in the VUMC admin UI. Leaving as is.